src/d3a/models/strategy/area_agents/one_sided_engine.py

```diff
@@ -143,7 +143,13 @@ def event_trade(self, *, trade):
                 f"offer: source_rate ({source_rate}) is not lower than target_rate ({target_rate})"
 
             try:
-                trade_offer_rate = trade.offer.energy_rate
+                if ConstSettings.IAASettings.MARKET_TYPE == 1 and trade is not None:
+                    # One sided market should subtract the fees
+                    trade_offer_rate = trade.offer.energy_rate - \
+                                       trade.fee_price / trade.offer.energy
+                else:
+                    # trade_offer_rate not used in two sided markets, trade_bid_info used instead
+                    trade_offer_rate = None
```

Comment:
really? not `trade_offer_rate = trade.offer.energy_rate` ?

Reply:
Ah yes, in the accept_offer, the offer rate is used then.

Reply:
It is not used at all in the 2-sided market indeed. The clause you mention is already handled by the market internally https://github.com/gridsingularity/d3a/blob/master/src/d3a/models/market/one_sided.py#L214